core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -470,11 +470,6 @@ protected void doSample() {
                         Transport.Connection connectionToClose = null;
 
                         @Override
-                        public void onAfter() {
-                            IOUtils.closeWhileHandlingException(connectionToClose);
```

Comment:
we also need to close the connection in `public void onFailure(Exception e) {` since we might get rejected or something like this.